e2e_playwright/st_plotly_chart_test.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import pytest
+from playwright.sync_api import Page, expect
+
+from e2e_playwright.conftest import ImageCompareFunction
+
+
+# Only do chromium as this can create a lot of screenshots
+# there should be no differences between chrome and safari and firefox
+@pytest.mark.only_browser("chromium")
+def test_plotly_has_consistent_visuals(
+    themed_app: Page, assert_snapshot: ImageCompareFunction
+):
+    ST_PLOTLY_CHART = f"{ST_PLOTLY_CHART}-"
```

Comment:
I don't think this templating makes anything more readable -- if anything it makes things worse, and we're unlikely to ever want to change this prefix (and if we do it'll be a file-level search-and-replace so won't be difficult anyway)